### PR TITLE
#277: Resolve references to price types

### PR DESCRIPTION
**Pull request by ValeSauer**
This PR adds reference resolution to customTypes of prices

#### Relevant Issues
- https://github.com/commercetools/commercetools-sync-java/issues/277

#### Todo

- Tests
    - [x] Unit 
    - [x] Integration
- [x] Documentation
- [x] Add Release Notes entry.

### CODE REVIEW

**Comment by heshamMassoud** [comment]
Thanks @ValeSauer and congrats for your first PR here 🍾 🎉 

I reviewed your changes and it looks good. I added more commits that contain:
- Some refactoring to the unit tests you added and added new ones: ab06cc6, 34c6963, 7225166, 5ba7bf9, 2406d85, 3f3cc8c
- updating javadocs: f368140
- fixed some typos in RELEASE_NOTES.md: 6ddc652
- prepare the new release notes and readme for the new version: b0ffc5f

@ValeSauer @lojzatran can you please review the changes.

File: src/integration-test/java/com/commercetools/sync/integration/commons/utils/ProductITUtils.java [hunk 1]
```diff
@@ -113,33 +120,63 @@ public static void deleteAllProducts(@Nonnull final SphereClient ctpClient) {
...
                 return ProductVariantDraftBuilder.of(productVariant)
                                                  .prices(priceDraftsWithChannelReferences)
                                                  .build();
             })
             .collect(toList());
         final List<PriceDraft> masterVariantPriceDrafts = productDraft
-            .getMasterVariant().getPrices().stream().map(price -> PriceDraftBuilder.of(price)
-                                                                                   .channel(channelReference)
-                                                                                   .build()).collect(toList());
+            .getMasterVariant().getPrices().stream()
```
Comment (lojzatran) [inline src/integration-test/java/com/commercetools/sync/integration/commons/utils/ProductITUtils.java]: Maybe this common part you can put it in a method:

```java
productVariant.getPrices().stream()
            .map(PriceDraftBuilder::of)
            .map(priceDraftBuilder -> ofNullable(channelReference).map(priceDraftBuilder::channel)
                                                                  .orElse(priceDraftBuilder))
            .map(priceDraftBuilder -> ofNullable(customFieldsDraft).map(priceDraftBuilder::custom)
                                                                   .orElse(priceDraftBuilder))
            .map(PriceDraftBuilder::build)
            .collect(toList())
```

File: src/main/java/com/commercetools/sync/products/utils/VariantReferenceReplacementUtils.java [hunk 2]
```diff
@@ -65,21 +67,27 @@
...
-     * returns as a result a {@link List} of {@link PriceDraft} that has all channel references with keys replacing the
-     * ids.
+     * Takes a product variant that is supposed to have all its prices' references (channel and custom type reference)
+     * expanded in order to be able to fetch the keys and replace the reference ids with the corresponding keys for the
+     * references. This method returns as a result a {@link List} of {@link PriceDraft} that has all channel and custom
+     * type references with keys replacing the ids.
      *
-     * <p>Any channel reference that is not expanded will have it's id in place and not replaced by the key.
+     * <p>Any reference, whether {@link Channel} or custom {@link io.sphere.sdk.types.Type}, that is not expanded will
+     * have it's id in place and not replaced by the key.
```
Comment (lojzatran) [inline src/main/java/com/commercetools/sync/products/utils/VariantReferenceReplacementUtils.java]: it's -> its

Comment (heshamMassoud) [reply src/main/java/com/commercetools/sync/products/utils/VariantReferenceReplacementUtils.java]: 8682be7

File: src/integration-test/java/com/commercetools/sync/integration/commons/utils/ProductITUtils.java (hunk 1 above)
Comment (heshamMassoud) [reply src/integration-test/java/com/commercetools/sync/integration/commons/utils/ProductITUtils.java]: 6619e70

**Pull request merged**